subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -394,6 +394,38 @@ static int register_ept(const struct device *instance, void **token,
 	return 0;
 }
 
+static int deregister_ept(const struct device *instance, void *token)
+{
+	struct backend_data_t *data = instance->data;
+	struct ipc_rpmsg_ept *rpmsg_ept;
+	int err;
+
+	/* Instance is not ready */
+	if (atomic_get(&data->state) != STATE_INITED) {
+		return -EBUSY;
+	}
+
+	rpmsg_ept = (struct ipc_rpmsg_ept *) token;
+
+	/* Endpoint is not registered with instance */
+	if (!rpmsg_ept) {
+		return -ENOENT;
+	}
+
+	if (data->role == ROLE_HOST) {
+		rpmsg_destroy_ept(&rpmsg_ept->ep);
+	} else {
+		err = ipc_rpmsg_deregister_ept(&data->rpmsg_inst, RPMSG_REMOTE, rpmsg_ept);
+		if (err != 0) {
+			return err;
+		}
+	}
```

Comment:
If you are HOST you are calling `rpmsg_destroy_ept` here. If you are REMOTE you are calling it from inside `ipc_rpmsg_deregister_ept`. Why?

Reply:
This designed to mirror the existing `register_ept` functions - as currently implemented, `rpmsg_create_ept` is only called from `ipc_rpmsg_register_ept` if you are a `REMOTE` and otherwise it is called within `ipc_rpmsg_static_vrings.c`

Reply:
Have removed the `ipc_rpmsg_deregister_ept` function and made a direct call into `rpmsg_destroy_ept`